examples/multi_gpu/papers100m_multigpu.py

```diff
@@ -0,0 +1,136 @@
+import argparse
+import os
+import time
+
+import torch
+import torch.distributed as dist
+import torch.multiprocessing as mp
+import torch.nn.functional as F
+from ogb.nodeproppred import PygNodePropPredDataset
+from torch.nn.parallel import DistributedDataParallel
+from torchmetrics import Accuracy
+
+from torch_geometric.loader import NeighborLoader
+from torch_geometric.nn import GCNConv
+
+
+def pyg_num_work():
+    num_work = None
+    if hasattr(os, "sched_getaffinity"):
+        try:
+            num_work = len(os.sched_getaffinity(0)) / 2
```

Comment:
n00b question: I think I didn't comment this in the previous PR, but why was this divided by 2?

Reply:
@rusty1s  gave me this hueristic a while back when I was setting up our GNN training infra inside NVIDIA and ive used it ever since. My hunch is that this is the most workers you can use at once without hurting perf for other processes due to busy cpu